assets/js/hooks/cell.js

```diff
@@ -215,6 +211,16 @@ const Cell = {
       }, 0);
     });
 
+    liveEditor.onViewUpdate((viewUpdate) => {
+      // We defer the check to happen after all focus/click events have
+      // been processed, in case the state changes as a result
+      setTimeout(() => {
+        if (this.isFocused && viewUpdate.selectionSet) {
```

Comment:
We can check if the selection changed:

```
!update.state.selection.eq(update.startState.selection)
```

I would make the event more specific `liveEditor.onSelectionChange` and have that check there.

The only difference is that when the user goes into insert mode in existing cell, the selection doesn't change, so we can call `sendCursorHistory` on focus separately.

Once we do that, I don't think we need the `setTimeout` here (since it's only specific to the focus case?). wdyt?

Reply:
I'm not sure if I understood about the focus change, because it makes the pubsub event to be dispatched twice